Objects/typeobject.c

```diff
@@ -2846,15 +2846,24 @@ static const short slotoffsets[] = {
 PyObject *
 PyType_FromSpecWithBases(PyType_Spec *spec, PyObject *bases)
 {
-    PyHeapTypeObject *res = (PyHeapTypeObject*)PyType_GenericAlloc(&PyType_Type, 0);
-    PyTypeObject *type, *base;
+    PyHeapTypeObject *res;
+    PyMemberDef *memb;
     PyObject *modname;
-    char *s;
-    char *res_start = (char*)res;
+    PyTypeObject *type, *base;
+
     PyType_Slot *slot;
+    Py_ssize_t nslots;
+    char *s, *res_start;
 
+    nslots = 0;
+    for (slot = spec->slots; slot->slot; slot++)
+        if (slot->slot == Py_tp_members)
+            for (memb = slot->pfunc; memb->name != NULL; memb++)
+                nslots++;
```

Comment:
Regarding style: In new code, please use braces around bodies of for & if statements.